File: 0617. 合并二叉树/递归.js
// 128ms
var mergeTrees = function (t1, t2) {
    if (t1 != null || t2 != null) {
        let node = new TreeNode((t1 == null ? 0 : t1.val) + (t2 == null ? 0 : t2.val))
        node.left = mergeTrees((t1 == null ? null : t1.left), (t2 == null ? null : t2.left))
        node.right = mergeTrees((t1 == null ? null : t1.right), (t2 == null ? null : t2.right))
        return node
    } else {
        return null
    }
};

File: 0872. 叶子相似的树/递归.swift
// 8ms
class Solution {
    func leafSimilar(_ root1: TreeNode?, _ root2: TreeNode?) -> Bool {
        func getLeafList(_ root: TreeNode?) -> [Int] {
            var result: [Int] = []
            func search(_ node: TreeNode?) {
                if let node = node {
                    if node.left != nil || node.right != nil {
                        search(node.left)
                        search(node.right)
                    } else {
                        result.append(node.val)
                    }
                }
            }
            search(root)
            return result
        }
        return getLeafList(root1) == getLeafList(root2)
    }
}